centaur/src/main/scala/centaur/CromwellTracker.scala

```diff
@@ -0,0 +1,63 @@
+package centaur
+
+import centaur.CromwellTracker._
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.api.model.WorkflowMetadata
+import cromwell.core.WorkflowProcessingEvent
+import cromwell.core.WorkflowProcessingEvents.DescriptionEventValue
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto._
+import io.circe.optics.JsonPath._
+import io.circe.parser.parse
+import io.circe.syntax._
+import org.apache.commons.math3.stat.inference.ChiSquareTest
+
+import scala.language.postfixOps
+
+
+case class CromwellTracker(backendCount: Int, configuredSignificance: Double) extends StrictLogging {
+  var counts: Map[String, Int] = Map()
+  def track(metadata: WorkflowMetadata): Unit = {
+
+    // Get all the workflow processing events for this workflow and pick out the "PickedUp" events. There can be multiple of
+    // these for a workflow if Cromwell was restarted while it was running.
+    val pickupEvents = parse(metadata.value) map {
+      root.workflowProcessingEvents.each.as[WorkflowProcessingEvent].getAll(_)
+    }
+
+    pickupEvents match {
+      case Left(exception) => throw exception
+      case Right(events) =>
+        events filter { _.description == DescriptionEventValue.PickedUp.value } foreach { e =>
+          synchronized {
+            counts = counts + (e.cromwellId -> (counts.getOrElse(e.cromwellId, 0) + 1))
+            logger.info("added count for Cromwell id {}: {}", e.cromwellId, counts(e.cromwellId))
+          }
+        }
+    }
+  }
+
+  def assertHoricromtality(): Unit = {
+    counts foreach { case (id, count) => logger.info(s"horicromtal count {}: {}", id, count) }
+
+    val runs = counts.values.sum
+    val expected: Array[Double] = (1 to backendCount) map { _ => runs.toDouble / backendCount } toArray
+    val actual: Array[Long] = counts.values map { _.toLong } toArray
+
+    val observedSignificance = new ChiSquareTest().chiSquareTest(expected, actual)
+    logger.info(f"configured/observed horicromtal significance levels: $configuredSignificance%.4f/$observedSignificance%.4f", configuredSignificance, observedSignificance)
+
+    if (observedSignificance < configuredSignificance) {
```

Comment:
Very cool!

How often are we likely to get random anomalies with the current significance? Not a PR-level concern, just curious how you reasoned about it.

Reply:
I used the customary p < 0.05 significance for now. N is pretty larger here (> 300) and in practice the lowest significance I've seen was > 0.42. As long Cromwell really is running horizontally I don't think we should have random failures.